python/ql/lib/semmle/python/ApiGraphs.qll

```diff
@@ -92,6 +117,37 @@ module API {
      */
     Node getReturn() { result = this.getASuccessor(Label::return()) }
 
+    /**
+     * Gets a node representing the `i`th parameter of the function represented by this node.
+     *
+     * This predicate may have multiple results when there are multiple invocations of this API component.
+     * Consider using `getAnInvocation()` if there is a need to distingiush between individual calls.
+     */
+    Node getParameter(int i) { result = this.getASuccessor(Label::parameter(i)) }
+
+    /**
+     * Gets the node representing the keyword parameter `name` of the function represented by this node.
+     *
+     * This predicate may have multiple results when there are multiple invocations of this API component.
+     * Consider using `getAnInvocation()` if there is a need to distingiush between individual calls.
+     */
+    Node getKeywordParameter(string name) {
+      result = this.getASuccessor(Label::keywordParameter(name))
+    }
+
+    /**
+     * Gets the number of parameters of the function represented by this node.
+     */
+    int getNumParameter() { result = max(int s | exists(this.getParameter(s))) + 1 }
+
+    /**
+     * Gets a node representing the last parameter of the function represented by this node.
+     *
+     * This predicate may have multiple results when there are multiple invocations of this API component.
+     * Consider using `getACall()` if there is a need to distingiush between individual calls.
+     */
+    Node getLastParameter() { result = this.getParameter(this.getNumParameter() - 1) }
```

Comment:
I copied it from the JavaScript implementation, it might not be useful for Python (and in that case we should delete it).  

In JS the last argument is often special. The last argument is often a callback or an options object in a function that uses a dynamic number of positional arguments.  
If that is not a pattern you can recognize for Python, then we should just delete this utility predicate.

Reply:
@RasmusWL is this something you have come across?

Reply:
I can not recall needing this, so I think we should delete it :+1:

Reply:
Ok, thanks :-)